app/src/main/java/com/winlator/core/TarCompressorUtils.java

```diff
@@ -171,7 +171,16 @@ private static boolean extract(Type type, InputStream source, File destination,
             TarArchiveEntry entry;
             while ((entry = (TarArchiveEntry)tar.getNextEntry()) != null) {
                 if (!tar.canReadEntryData(entry)) continue;
-                File file = new File(destination, entry.getName());
+
+                // Skip macOS metadata files (._{filename}) and __MACOSX directories
```

Comment:
An annoying and ridiculous issue, but one we need to watch out for when doing extractions. The extraction algorithm can break if there's macOSX metadata files, so this avoids it.

I really dislike it, but it's the cleanest way to get around it without reworking the extraction algorithm.

Reply:
Hoping this doesn't break other packages. A lot have been packaged on my mac 😅 so let's keep an eye on this and make sure fresh installs work etc